pkg/endpoint/policy.go

```diff
@@ -809,6 +812,25 @@ func (e *Endpoint) TriggerPolicyUpdatesLocked(owner Owner, opts models.Configura
 	return changed, ctCleaned, nil
 }
 
+func (e *Endpoint) runIdentityToK8sPodSync() {
+	e.controllers.UpdateController("sync-identity-to-k8s-pod",
+		controller.ControllerParams{
+			DoFunc: func() error {
+				e.Mutex.RLock()
+				id := e.SecLabel.ID.String()
```

Comment:
Just a small concern here, we have parts of our code where we check for `e.SecLabel != nil`. Shouldn't we check it here as well?

Reply:
Good point, we should.